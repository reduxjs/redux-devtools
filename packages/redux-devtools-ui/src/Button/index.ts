export { default } from './Button.js';
