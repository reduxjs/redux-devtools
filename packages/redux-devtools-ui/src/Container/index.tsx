import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { useTheme, ThemeData } from '../utils/theme';
import { MainContainerWrapper, ContainerWrapper } from './styles';
import { Theme } from '../themes/default';

interface ContainerFromThemeDataProps {
  children?: React.ReactNode;
  themeData: ThemeData;
  className?: string;
}

const ContainerFromThemeData: React.FunctionComponent<
  ContainerFromThemeDataProps
> = ({ themeData, className, children }) => {
  const theme = useTheme(themeData);
  return (
    <ThemeProvider theme={theme}>
      <MainContainerWrapper className={className}>
        {children}
      </MainContainerWrapper>
    </ThemeProvider>
  );
};

interface Props {
  children?: React.ReactNode;
  themeData?: ThemeData;
  theme?: Theme;
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({
  themeData,
  className,
  theme,
  children,
}) => {
  if (!themeData) {
    return (
      <ContainerWrapper className={className} theme={theme}>
        {children}
      </ContainerWrapper>
    );
  }

  return (
    <ContainerFromThemeData themeData={themeData} className={className}>
      {children}
    </ContainerFromThemeData>
  );
};

Container.propTypes = {
  children: PropTypes.node,
  themeData: PropTypes.any,
  theme: PropTypes.any,
  className: PropTypes.string,
};

export default Container;
