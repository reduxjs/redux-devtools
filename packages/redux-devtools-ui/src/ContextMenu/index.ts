export { default } from './ContextMenu.js';
