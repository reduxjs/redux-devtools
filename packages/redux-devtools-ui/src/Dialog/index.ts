export { default } from './Dialog.js';
