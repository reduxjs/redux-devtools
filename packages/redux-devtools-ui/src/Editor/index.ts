export { default } from './Editor.js';
