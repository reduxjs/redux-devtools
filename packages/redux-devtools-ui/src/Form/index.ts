export { default } from './Form.js';
