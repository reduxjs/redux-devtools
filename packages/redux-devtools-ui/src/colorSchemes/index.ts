export { default } from './default';
export { default as github } from './github';
export { default as atomOneDark } from './atom-one-dark';
export { default as dracula } from './dracula';
export { default as iRBlack } from './ir-black';
export { default as macintosh } from './macintosh';
export { default as materia } from './materia';
export { default as oceanicNext } from './oceanic-next';
export { default as phD } from './phd';
export { default as pico } from './pico';
export { default as solarFlare } from './solar-flare';
export { default as spacemacs } from './spacemacs';
export { default as unikitty } from './unikitty';
export { default as woodland } from './woodland';
