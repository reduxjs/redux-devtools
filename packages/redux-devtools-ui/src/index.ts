export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as ContextMenu } from './ContextMenu';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs, Tab } from './Tabs';
export { default as SegmentedControl } from './SegmentedControl';
export { default as Notification } from './Notification';
export * from './Toolbar';

import color from './utils/color';
export const effects = { color };
export { default as createStyledComponent } from './utils/createStyledComponent';
export { Theme, ThemeFromProvider, Scheme } from './utils/theme';
