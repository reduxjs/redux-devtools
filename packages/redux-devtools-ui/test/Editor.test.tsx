import React from 'react';
import { render } from '@testing-library/react';
import { Editor } from '../src';

describe('Editor', function () {
  const { container } = render(<Editor value="var a = 1;" />);

  it('renders correctly', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
