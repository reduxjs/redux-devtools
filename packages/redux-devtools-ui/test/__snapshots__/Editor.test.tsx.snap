// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor renders correctly 1`] = `
<div
  class="sc-dhKdcB cZUDgE"
>
  <div
    class="CodeMirror cm-s-default"
    translate="no"
  >
    <div
      style="overflow: hidden; position: relative; width: 3px; height: 0px;"
    >
      <textarea
        autocapitalize="off"
        autocorrect="off"
        spellcheck="false"
        style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; min-height: 1em; outline: none;"
        tabindex="0"
      />
    </div>
    <div
      class="CodeMirror-vscrollbar"
      cm-not-content="true"
      tabindex="-1"
    >
      <div
        style="min-width: 1px;"
      />
    </div>
    <div
      class="CodeMirror-hscrollbar"
      cm-not-content="true"
      tabindex="-1"
    >
      <div
        style="height: 100%; min-height: 1px;"
      />
    </div>
    <div
      class="CodeMirror-scrollbar-filler"
      cm-not-content="true"
    />
    <div
      class="CodeMirror-gutter-filler"
      cm-not-content="true"
    />
    <div
      class="CodeMirror-scroll"
      tabindex="-1"
    >
      <div
        class="CodeMirror-sizer"
        style="margin-left: 0px; min-width: 3px;"
      >
        <div
          style="position: relative;"
        >
          <div
            class="CodeMirror-lines"
            role="presentation"
          >
            <div
              role="presentation"
              style="position: relative; outline: none;"
            >
              <div
                class="CodeMirror-measure"
              />
              <div
                class="CodeMirror-measure"
              >
                <pre
                  class="CodeMirror-line"
                  role="presentation"
                >
                  <span
                    role="presentation"
                    style="padding-right: .1px;"
                  >
                    <span
                      class="cm-keyword"
                    >
                      var
                    </span>
                     
                    <span
                      class="cm-def"
                    >
                      a
                    </span>
                     
                    <span
                      class="cm-operator"
                    >
                      =
                    </span>
                     
                    <span
                      class="cm-number"
                    >
                      1
                    </span>
                    ;
                  </span>
                </pre>
              </div>
              <div
                style="position: relative; z-index: 1;"
              />
              <div
                class="CodeMirror-cursors"
              />
              <div
                class="CodeMirror-code"
                role="presentation"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        style="position: absolute; height: 50px; width: 1px;"
      />
      <div
        class="CodeMirror-gutters"
      >
        <div
          class="CodeMirror-gutter CodeMirror-linenumbers"
          style="width: 1px;"
        />
        <div
          class="CodeMirror-gutter CodeMirror-foldgutter"
        />
      </div>
    </div>
  </div>
</div>
`;
